### DISCUSSION TITLE
Print few lines in .pyx file

### DISCUSSION THREAD

**Discussion by rkoystart** [explosion/spaCy#11447] [Help: Coding & Implementations] [answered]
Hi, wanted to print few lines in doc.pyx files. So tried using the following print-statement

`print("my required message", object)`

and the above didnt work. Can anyone let me know how to provide a proper print statement and verify things.
And also have few doubts regarding .pyx, .pyi, .pyd files. Can anyone give me a breifing about what these files are actually used instead of .py files

**Comment by polm** [comment, marked as answer]
All of those files are Cython. Depending on the context in Cython, `print` can work as it does in normal Python or pretty differently. Can you give an actual bit of sample code and the error?

https://cython.org/

**Reply by rkoystart** [reply]
Hi @polm , Thanks for the response. 
I tried adding the following line in the contructor(__init__()) of the file doc.pyx in the path env/lib/python3.8/site-packages/spacy/tokens/doc.pyx
`
print("This is the contructor of the class doc")`

and when i executed the following code in the terminal

```
import spacy
model = spacy.load("en_core_web_md")
doc = model("hello how are you")
```
I didnt get any print message i had added previously.

**Reply by polm** [reply]
Cython needs to be compiled before you can use it. Take a look at the section on [building spaCy](https://github.com/explosion/spaCy#-compile-from-source), and note that after changes to Cython files you have to run `python setup.py build_ext --inplace`. 

If it's your first time working with Cython it might help to go through a Cython tutorial first.
